src/stores/artists-store.js

```diff
@@ -0,0 +1,17 @@
+import { autorun, observable } from 'mobx'
+
+class ArtistsStore {
+  @observable artists = []
+  @observable searchedArtist = ''
+  @observable foundArtists = []
+}
+
+const artistsStore = new ArtistsStore()
+
+export default artistsStore
+
+autorun(() => {
```

Comment:
Imho lepiej zrobić metodę `loadArtists` albo coś takiego, w której robisz requesta do API, obcinasz tylko do 5 artystów (chociaż widziałem, że masz to jako paramter do requesta) i aktualizujesz `foundArtists`. Zamiast mutować `foundArtists` lepiej nadpisać całkowicie nową `this.foundArtists = res.artists` czy coś takiego. Wtedy nie będziesz miał problemów z duplikatami i tym, że masz więcej niż 5 artystów.

Reply:
zrobilem tak, ze stworzylem actiona w storze, ktory czysci foundArtists i leci przed mapowaniem nowych. Juz sie nie buguje, ale teraz jak zwraca mi jednego artista, to poprzedni znikaja. W sumie niby dobrze, ale wczesniej zostawly mi 'mniej dokladne' trafienia artystow, co tez bylo spoko 😕

Reply:
Hmm możesz takie coś zrobić, że zawsze wyświetla 5. Wymagałoby trochę więcej logiki, żeby nie było duplikatów i nie przekraczało 5 artystów, ale do zrobienia :P

Reply:
zobacze jak sie sprawdza to rozwiazania i jesli bede chcial wrocic do wyswietlania wiekszej ilosci ziomeczkow, to bede kombinowac. Poki co po tym PR chce dorobic wyswietlanie info o artyscie na klik :p